modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -1002,6 +1005,44 @@ public boolean created() {
     @Override protected CorruptedTreeException corruptedTreeException(String msg, Throwable cause, int grpId, long... pageIds) {
         CorruptedTreeException e = new CorruptedTreeException(msg, cause, grpName, cacheName, idxName, grpId, pageIds);
 
+        String errorMsg = "Index " + idxName + " of the table " + tblName + " is corrupted, to fix this issue a " +
+            "rebuild is required. On the next restart, node will enter the maintenance mode and rebuild corrupted " +
+            "indexes.";
+
+        log.warning(errorMsg);
+
+        int cacheId = table.cacheId();
+
+        try {
+            String description = "Corrupted index found";
+
+            MaintenanceTask task = new MaintenanceTask(INDEX_REBUILD_MNTC_TASK_NAME,
+                description,
+                cacheId + INDEX_REBUILD_PARAMETER_SEPARATOR + idxName
```

Comment:
Let's add a method something like **MaintenanceRebuildIndexTarget#toMaintenanceTaskParameters** that returns a string of parameters. 

Also, let's add a test that checks this two **MaintenanceRebuildIndexTarget#toMaintenanceTaskParameters** and **MaintenanceRebuildIndexTarget#parseMaintenanceTaskParameters** methods and that they will work with each other.

Could you check that the cache and index names cannot contain a **INDEX_REBUILD_PARAMETER_SEPARATOR**?